fields/types/relationship/RelationshipFieldForTag.js

```diff
@@ -0,0 +1,240 @@
+    return this.props.collapse && !this.props.value;
+  },
+
+  buildFilters() {
+    var filters = {};
+
+    _.each(this.props.filters, (value, key) => {
+      if (_.isString(value) && value[0] == ':') { // eslint-disable-line eqeqeq
+        var fieldName = value.slice(1);
+
+        var val = this.props.values[fieldName];
+        if (val) {
+          filters[key] = val;
+          return;
+        }
+
+        // check if filtering by id and item was already saved
+        if (fieldName === ':_id' && Keystone.item) {
+          filters[key] = Keystone.item.id;
+          return;
+        }
+      } else {
+        filters[key] = value;
+      }
+    }, this);
+
+    var parts = [];
+
+    _.each(filters, function(val, key) {
+      parts.push('filters[' + key + '][value]=' + encodeURIComponent(val));
+    });
+
+    return parts.join('&');
+  },
+
+  cacheItem(item) {
+    // Filter out new version tags for old/new compatible
+    if (item && NEW_SUBCATEGORY_IDS.includes(item.id)) {
+      return;
+    }
+    item.href = Keystone.adminPath + '/' + this.props.refList.path + '/' + item.id;
+    this._itemsCache[item.id] = item;
+  },
+
+  loadValue(values) {
+    if (!values) {
+      return this.setState({
+        loading: false,
+        value: null,
+      });
+    };
+    values = Array.isArray(values) ? values : values.split(',');
+    let cachedValues = values.map(i => this._itemsCache[i]).filter(i => i);
+    if (cachedValues.length === values.length) {
+      this.setState({
+        loading: false,
+        value: this.props.many ? cachedValues : cachedValues[0],
+      });
+      return;
+    }
+    this.setState({
+      loading: true,
+      value: null,
+    });
+    async.map(values, (value, done) => {
+      xhr({
+        url: Keystone.adminPath + '/api/' + this.props.refList.path + '/' + value + '?basic',
+        responseType: 'json',
+      }, (err, resp, data) => {
+        if (err || !data) return done(err);
+        this.cacheItem(data);
+        done(err, data);
+      });
+    }, (err, expanded) => {
+      if (!this.isMounted()) return;
+      this.setState({
+        loading: false,
+        value: this.props.many ? expanded : expanded[0],
+      });
+    });
+  },
+
+  // NOTE: this seems like the wrong way to add options to the Select
+  loadOptionsCallback: {},
+  loadOptions(input, callback) {
+    // NOTE: this seems like the wrong way to add options to the Select
+    this.loadOptionsCallback = callback;
+    let filters = this.buildFilters();
+    xhr({
+      url: Keystone.adminPath + '/api/' + this.props.refList.path + '?basic&search=' + input + '&' + filters,
+      responseType: 'json',
+    }, (err, resp, data) => {
+      if (err) {
+        console.error('Error loading items:', err);
+        return callback(null, []);
+      }
+      data.results.forEach(this.cacheItem);
+      callback(null, {
+        options: data.results.filter(subcategory => !NEW_SUBCATEGORY_IDS.includes(subcategory.id)),
```

Comment:
The feature that `tag` options would exclude those tag marked as `subcategory` will need to be obtain in the future (even afrer new info structure release).
For the future need, would it be better if we change the filter to filter out tags that has `category` field ?

Reply:
unused fields (such as `category` field) is automatically filtered out by keystone, so we dont have such fields to obtain here

![Screen Shot 2022-10-03 at 2 40 12 PM](https://user-images.githubusercontent.com/25971696/193515458-434545e7-3e17-4178-a48f-443b40dfba3e.png)

Reply:
Got it !
I think this problem could be wait until developing category set edition in the future.